sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -31,7 +31,12 @@ class JDBCOptions(
 
   import JDBCOptions._
 
-  def this(parameters: Map[String, String]) = this(new CaseInsensitiveMap(parameters))
+  def this(params: Map[String, String]) = {
+    this(params match {
+      case cMap: CaseInsensitiveMap => cMap
+      case _ => new CaseInsensitiveMap(params)
```

Comment:
Moved it to CaseInsenstiveMap and marked the constructor private to avoid any nested created of the case-insensitive maps.